### ISSUE TITLE
Modifying a signed file still passes validation

### ISSUE THREAD

**Issue by 123cdxcc** [digitorus/pdfsign#47]
English is not my native language, I can express may not be clear, I will provide the code!    
Thank you very much if you can help me out!   
I am currently using the library for the signature of the pdf file, but when I have signed the pdf file at the end of the increase in bytes of data, and then go to the checksum, or checksum through, I do not know whether this is consistent with it does not seem to detect that the document has been tampered with!    
I was able to detect in adobe reader that the file had been tampered with.   

signature function:
```go
//  signature
func signature(input, output string) error {
	input_file, err := os.Open(input)
	if err != nil {
		panic(err)
	}
	defer input_file.Close()

	output_file, err := os.Create(output)
	if err != nil {
		panic(err)
	}
	defer output_file.Close()

	finfo, err := input_file.Stat()
	if err != nil {
		panic(err)
	}
	size := finfo.Size()

	rdr, err := pdf.NewReader(input_file, size)
	if err != nil {
		panic(err)
	}

	certificate_data, err := os.ReadFile("certificate.crt")
	if err != nil {
		panic(err)
	}
	certificate_data_block, _ := pem.Decode(certificate_data)
	if certificate_data_block == nil {
		//log.Fatal(errors.New("failed to parse PEM block containing the certificate"))

	}

	cert, err := x509.ParseCertificate(certificate_data_block.Bytes)
	if err != nil {
		panic(err)
	}

	privateKeyFs, err := os.ReadFile("private_key.pem")
	if err != nil {
		panic(err)
	}
	key_data_block, _ := pem.Decode(privateKeyFs)
	if key_data_block == nil {
		panic(errors.New("failed to parse PEM block containing the private key"))
	}
	pkey, err := x509.ParsePKCS1PrivateKey(key_data_block.Bytes)
	if err != nil {
		var t any
		t, err = x509.ParsePKCS8PrivateKey(key_data_block.Bytes)
		pkey = t.(*rsa.PrivateKey)
		if err != nil {
			panic(err)
		}
	}

	certificate_chains := make([][]*x509.Certificate, 0)
	// pass:123456
	err = sign.Sign(input_file, output_file, rdr, size, sign.SignData{
		Signature: sign.SignDataSignature{
			Info: sign.SignDataSignatureInfo{
				Name:        "xxxx",
				Location:    "xxxx",
				Reason:      "xxxx",
				ContactInfo: "xxxx",
				Date:        time.Now().Local(),
			},
			CertType:   sign.CertificationSignature,
			DocMDPPerm: sign.AllowFillingExistingFormFieldsAndSignaturesPerms,
		},
		Signer:            pkey,               // crypto.Signer
		DigestAlgorithm:   crypto.SHA256,      // hash algorithm for the digest creation
		Certificate:       cert,               // x509.Certificate
		CertificateChains: certificate_chains, // x509.Certificate.Verify()
		TSA: sign.TSA{
			URL:      "https://freetsa.org/tsr",
			Username: "",
			Password: "",
		},

		// The follow options are likely to change in a future release
		//
		// cache revocation data when bulk signing
		RevocationData: revocation.InfoArchival{},
		// custom revocation lookup
		RevocationFunction: sign.DefaultEmbedRevocationStatusFunction,
	})
	if err != nil {
		panic(err)
	} else {
		log.Println("Signed PDF written to " + output)
	}
	return nil
}
```
verify function:
```go
// verify
func verifyPdf(pdfName string) {
	input_file, err := os.Open(pdfName)
	if err != nil {
		panic(err)
	}
	defer input_file.Close()

	resp, err := verify.File(input_file)
	if err != nil {
		panic(err)
	}
	jsonData, err := json.MarshalIndent(resp, "", "\t")
	if err != nil {
		panic(err)
	}
	// 将jsonData的数据写入文件
	err = os.WriteFile("verify.json", jsonData, 0644)
	//fmt.Println(string(jsonData))
	return
}
```
main function:
```go
// call signing
err = signature("a.pdf", "b.pdf")
if err != nil {
  panic(err)
}
// tamper
data, err := os.ReadFile("b.pdf")
if err != nil {
  panic(err)
}
data = append(data, []byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f}...)
// call verify 
err = os.WriteFile("c.pdf", data, 0644)
if err != nil {
  panic(err)
}
verifyPdf("c.pdf")
```

**Comment by vanbroup** [comment]
You are appending data, which should make the PDF invalid but not the signature which is bound to the document.

Please try to modify by data within the PDF instead.

**Comment by 123cdxcc** [comment]
> You are appending data, which should make the PDF invalid but not the signature which is bound to the document.
> 
> Please try to modify by data within the PDF instead.

Attaching data to the end does not invalidate the PDF, and the file still opens normally.
When I use the PDF reader of mac to open the file and modify the file, adobe reader will show that I have modified the PDF, and can also be specific to the page, but I check in pdfsign is still passed.
It seems that the verification of pdfsign can only verify the integrity of the signature, and can not verify whether the PDF is modified.
After the verification, the signature information is displayed. However, when I use the public key to decrypt the signature, the decryption fails. Or use the PDF file hash before the signature to compare but both indicate that I failed：crypto/rsa: verification error
